.jenkins/Jenkinsfile

```diff
@@ -11,762 +11,518 @@
-                  if (system_keyspace_directory.result != 'FAILURE') break
-              }
-              if (system_keyspace_directory.result != 'SUCCESS') unstable('system-keyspace-directory failures')
-              if (system_keyspace_directory.result == 'FAILURE') currentBuild.result='FAILURE'
-            }
-          }
-          post {
-            always {
-                warnError('missing test xml files') {
-                    script {
-                        copyTestResults('test-system-keyspace-directory', system_keyspace_directory.getNumber())
-                    }
-                }
-            }
-          }
+    stage('Summary') {
+      steps {
+        summateStatus()
+      }
+    }
+  }
+  post {
+    always {
+      generateTestReports()
+      sendNotifications()
+    }
+  }
+}
+
+///////////////////////////
+//// scripting support ////
+///////////////////////////
+
+def jdksSupported() { return ["11", "17"] }
+def archsSupported() { return ["amd64", "arm64"] }
+def pythonsSupported() { return ["3.8", "3.11"] }
+def jdkDefault() { return "11" }
+def pythonDefault() { return "3.8" }
+
+def pipelineProfiles() {
+    return [
+        'packaging': ['artifacts', 'lint', 'debian', 'redhat'],
+        'skinny': ['lint', 'cqlsh-test', 'test', 'jvm-dtest', 'simulator-dtest', 'dtest'],
+        'pre-commit': ['artifacts', 'lint', 'debian', 'redhat', 'fqltool-test', 'cqlsh-test', 'test', 'test-latest', 'stress-test', 'test-burn', 'jvm-dtest', 'simulator-dtest', 'dtest', 'dtest-latest'],
+        'pre-commit w/ upgrades': ['artifacts', 'lint', 'debian', 'redhat', 'fqltool-test', 'cqlsh-test', 'test', 'test-latest', 'stress-test', 'test-burn', 'jvm-dtest', 'jvm-dtest-upgrade', 'simulator-dtest', 'dtest', 'dtest-novnode', 'dtest-latest', 'dtest-upgrade'],
+        'post-commit': ['artifacts', 'lint', 'debian', 'redhat', 'fqltool-test', 'cqlsh-test', 'test-cdc', 'test', 'test-latest', 'test-compression', 'stress-test', 'test-burn', 'long-test', 'test-oa', 'test-system-keyspace-directory', 'jvm-dtest', 'jvm-dtest-upgrade', 'simulator-dtest', 'dtest', 'dtest-novnode', 'dtest-latest', 'dtest-large', 'dtest-large-novnode', 'dtest-upgrade', 'dtest-upgrade-novnode', 'dtest-upgrade-large', 'dtest-upgrade-novnode-large'],
+        'custom': []
+    ]
+}
+
+def tasks() {
+    // Steps config
+    def buildSteps = [
+      'jar': [script: 'build-jars.sh', toCopy: null],
+      'artifacts': [script: 'build-artifacts.sh', toCopy: 'apache-cassandra-*.tar.gz,apache-cassandra-*.jar,apache-cassandra-*.pom'],
+      'lint': [script: 'check-code.sh', toCopy: null],
+      'debian': [script: 'build-debian.sh', toCopy: 'cassandra_*,cassandra-tools_*'],
+      'redhat': [script: 'build-redhat.sh rpm', toCopy: '*.rpm'],
+    ]
+    buildSteps.each() {
+        it.value.put('type', 'build')
+        it.value.put('size', 'small')
+        it.value.put('splits', 1)
+    }
+
+    def testSteps = [
```

Comment:
I think we should add a `print` command to `run-tests.sh` (or `docker/run-tests.sh`) and populate this list from the output of that command.

Reply:
yes.

that will happen as part of [CASSANDRA-18731](https://issues.apache.org/jira/browse/CASSANDRA-18731).
(but it won't be run-tests.sh that is responsible for defining the list)